ukhsa.oas.rules.yml

```diff
@@ -264,3 +265,42 @@ rules:
     recommended: true
     description: '`201 Created` responses to `POST` methods SHOULD have a `Location` header identifying the location of the newly created resource.'
     message: '{{error}}'
+
+  # MUST Each operation MUST define 400, 404, and 500 responses using `application/problem+json` with examples.
+  # => /docs/spectral-rules/must/must-define-critical-problem-responses.md
+
+  must-define-critical-problem-responses:
+    description: "Operations must define 400, 404, 500 with examples and application/problem+json"
+    message: "{{error}}"
+    severity: warn
+    given: $.paths[*][*]
+    then:
+      function: has-required-problem-details-error-responses
+      functionOptions:
+        mode: 'critical'
+
+  # MUST Operations with their own non-empty `security` MUST define 401 and 403 responses using `application/problem+json` with examples.
+  # => /docs/spectral-rules/must/must-define-security-problem-responses.md
+
+  must-define-security-problem-responses:
+    description: "Operations with their own non-empty 'security' MUST define 401 and 403"
+    message: "{{error}}"
+    severity: warn
+    given: $.paths[*][*]
+    then:
+      function: has-required-problem-details-error-responses
+      functionOptions:
+        mode: 'explicit-security'
+
+  # SHOULD Root path SHOULD define 401 and 403 when inheriting global security, using `application/problem+json` with examples.
+  # => /docs/spectral-rules/should/should-define-root-inherited-security-responses.md
+
+  should-define-root-inherited-security-responses:
```

Comment:
might need to be removed based on the comment for the root path security requirements

Reply:
Removed in https://github.com/ukhsa-collaboration/api-guidelines/pull/42/commits/fd5169a07154021856954dfe28db914d80dda03e